integration/src/main/java/com/aws/greengrass/mqtt/moquette/BrokerKeyStore.java

```diff
@@ -75,10 +76,12 @@ public void initialize() throws KeyStoreException {
      * @throws KeyStoreException If unable to set key entry.
      */
     public void updateServerCertificate(CertificateUpdateEvent certificateUpdate) throws KeyStoreException {
-        // TODO: Support certificate chains
-        Certificate[] certChain = {certificateUpdate.getCertificate()};
-        jks.setKeyEntry(BROKER_KEY_ALIAS, certificateUpdate.getKeyPair().getPrivate(), jksPassword.toCharArray(),
-            certChain);
+        X509Certificate[] fullChain = Stream.concat(Stream.of(certificateUpdate.getCertificate()),
+                Stream.of(certificateUpdate.getCaCertificates()))
+            .toArray(X509Certificate[]::new);
+
+        jks.setKeyEntry(BROKER_KEY_ALIAS, certificateUpdate.getKeyPair().getPrivate(),
+            jksPassword.toCharArray(), fullChain);
```

Comment:
While we do want to omit the self signed root cert, I'm not finding a reliable way to achieve this without writing DER parsing code or pulling in big dependencies.
Need to re-think the approach here. It may make sense to add this as a config option in CDA itself (since it already has a dependency on BouncyCastle). Also, if we put it in CDA, then EMQX also benefits.